airflow/contrib/hooks/sagemaker_hook.py

```diff
@@ -0,0 +1,167 @@
+            self.is_key("S3DataSource", channel['DataSource'], dict_name="DataSource")
+            self.is_key("S3DataType", channel['DataSource']['S3DataSource'],
+                        dict_name="S3DataSource")
+            self.is_key("S3Uri", channel['DataSource']['S3DataSource'],
+                        dict_name="S3DataSource")
+        self.is_key("OutputDataConfig", config)
+        self.is_key("S3OutputPath", config['OutputDataConfig'],
+                    dict_name="OutputDataConfig")
+        self.is_key("ResourceConfig", config)
+        self.is_key("InstanceType", config['ResourceConfig'], "ResourceConfig")
+        self.is_key("InstanceCount", config['ResourceConfig'], "ResourceConfig")
+        self.is_key("VolumeSizeInGB", config['ResourceConfig'], "ResourceConfig")
+        if self.is_key("VpcConfig", config):
+            self.is_key("SecurityGroupIds", config['VpcConfig'], "VpcConfig")
+            self.is_key("Subnets", config['VpcConfig'], "VpcConfig")
+        self.is_key("StoppingCondition", config)
+        if self.is_key("Tags", config):
+            self.is_key("Key", config['Tags'], "Tags")
+            self.is_key("Value", config['Tags'], "Tags")
+        if not config['AlgorithmSpecification']['TrainingInputMode'] \
+                in ('Pipe', 'File'):
+            raise AirflowException("TrainingInputMode should either be Pipe or File.")
+        if not config['InputDataConfig']['DataSource']['S3DataSource']['S3DataType']\
+                in ('ManifestFile', 'S3Prefix'):
+            raise AirflowException("S3DataType should either be "
+                                   "ManifestFile or S3Prefix.")
+        if not config['ResourceConfig']['InstanceType'] in InstanceTypes:
+            raise AirflowException("Invalid InstanceType.")
+        self.checkt_for_url(config['InputDataConfig']
+                                  ['DataSource']['S3DataSource']['S3Uri'])
+
+    def get_conn(self):
+        self.conn = self.get_client_type('sagemaker')
+        return self.conn
+
+    def list_training_job(self, job_name):
+        sagemaker_conn = self.get_conn()
+        return sagemaker_conn.list_training_jobs(NameContains=job_name)
+
+    def list_tuning_job(self, job_name):
+        sagemaker_conn = self.get_conn()
+        return sagemaker_conn.list_hyper_parameter_tuning_job(NameContains=job_name)
+
+    def create_training_job(self, training_job_config):
+
+        if self.use_db_config:
+            if not self.sagemaker_conn_id:
+                raise AirflowException("SageMaker connection id must be present to read \
+                                        SageMaker training jobs configuration.")
+            sagemaker_conn = self.get_connection(self.sagemaker_conn_id)
+
+            config = sagemaker_conn.extra_dejson.copy()
+            tempconfig = config
+            training_job_config.update(tempconfig)
+
+        # run checks
```

Comment:
Didn't see the check method actually gets executed.

Reply:
Called the check method